apps/institution-service/src/test/institutionSearchAndFilter.test.ts

```diff
@@ -0,0 +1,332 @@
+    {
+      name: string;
+      aggregatorIntegrations: [
+        {
+          isActive: boolean;
+          supports_aggregation: boolean;
+          supports_identification: boolean;
+          supports_verification: boolean;
+          supports_history: boolean;
+          supports_oauth: boolean;
+          aggregator: {
+            name: string;
+          };
+        },
+      ];
+    },
+  ];
+}
+
+describe("get institutions with filters and search", () => {
+  it("gets a list of institutions including mx or finicity integrations", async () => {
+    const req = buildInstitutionRequest({
+      aggregatorName: ["mx", "finicity"],
+    });
+    const res = {
+      json: jest.fn(),
+      status: jest.fn().mockReturnThis(),
+    } as unknown as Response;
+
+    await getPaginatedInstitutions(req, res);
+
+    const jsonResponse =
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+      (res.json as jest.Mock).mock.calls[0][0] as PaginatedInstitutionResponse;
+    expect(res.status).toHaveBeenCalledWith(200);
+    expect(jsonResponse.currentPage).toBe(1);
+    expect(jsonResponse.totalRecords).toBeGreaterThan(0);
+    jsonResponse.institutions.forEach((institution) => {
+      let mxFound = false;
+      let finicityFound = false;
+      institution.aggregatorIntegrations.forEach((integration) => {
+        if (integration.aggregator.name === "mx") {
+          mxFound = true;
+        } else if (integration.aggregator.name === "finicity") {
+          finicityFound = true;
+        }
+      });
+      expect(mxFound || finicityFound).toBeTruthy();
+    });
+  });
+
+  it("gets a list of institutions that have an integration that supports filtered job types", async () => {
+    const req = buildInstitutionRequest({
+      supportsIdentification: true,
+      supportsVerification: true,
+      supportsAggregation: true,
+      supportsHistory: true,
+    });
+    const res = {
+      json: jest.fn(),
+      status: jest.fn().mockReturnThis(),
+    } as unknown as Response;
+
+    await getPaginatedInstitutions(req, res);
+
+    const jsonResponse =
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+      (res.json as jest.Mock).mock.calls[0][0] as PaginatedInstitutionResponse;
+    expect(res.status).toHaveBeenCalledWith(200);
+    expect(jsonResponse.totalRecords).toBeGreaterThan(0);
+    jsonResponse.institutions.forEach((institution) => {
+      let jobTypesSupported = false;
+      institution.aggregatorIntegrations.forEach((aggInt) => {
+        if (
+          aggInt.supports_aggregation &&
+          aggInt.supports_identification &&
+          aggInt.supports_verification &&
+          aggInt.supports_history
+        ) {
+          jobTypesSupported = true;
+        }
+      });
+      expect(jobTypesSupported).toBeTruthy();
+    });
+  });
+
+  it("institution list by default only includes institutions which have at least one active integration", async () => {
```

Comment:
I think we want institutions to show up even if they don't have an active integration. When we're filtering by an aggregator, then it would need be an active integration for that aggregator.

It would be confusing to create a new institution, and have it not show up in the list until you click the includeActiveIntegrations switch